code/modules/mob/living/carbon/xenomorph/abilities/praetorian/praetorian_powers.dm

```diff
@@ -867,3 +867,150 @@
+	if(!istype(BD))
+		return
+
+	if(X.observed_xeno != null)
+		to_chat(X, SPAN_XENOHIGHDANGER("You cannot retrieve sisters through overwatch!"))
+		return
+
+	if(!isXeno(A) || !X.can_not_harm(A))
+		to_chat(X, SPAN_XENODANGER("You must target one of your sisters!"))
+		return
+
+	if(A == X)
+		to_chat(X, SPAN_XENODANGER("You cannot retrieve yourself!"))
+		return
+
+	if(X.anchored)
+		to_chat(X, SPAN_XENODANGER("That sister cannot move!"))
+		return
+
+	if(!(A in view(7, X)))
+		to_chat(X, SPAN_XENODANGER("That sister is too far away!"))
+		return
+
+	var/mob/living/carbon/Xenomorph/targetXeno = A
+
+	if(!(targetXeno.resting || targetXeno.stat == UNCONSCIOUS))
+		if(targetXeno.mob_size > MOB_SIZE_BIG)
+			to_chat(X, SPAN_WARNING("[targetXeno] is too big to retrieve while standing up!"))
+			return
+
+	if(targetXeno.stat == DEAD)
+		to_chat(X, SPAN_WARNING("[targetXeno] is already dead!"))
+		return
+
+	if(!action_cooldown_check() || X.action_busy)
+		return
+
+	if(!X.check_state())
+		return
+
+	if(!check_plasma_owner())
+		return
+
+	if(!BD.use_internal_hp_ability(retrieve_cost))
+		return
+
+	if(!check_and_use_plasma_owner())
+		return
+
+	// Build our turflist
+	var/list/turf/turflist = list()
+	var/list/telegraph_atom_list = list()
+	var/facing = get_dir(X, A)
+	var/reversefacing = get_dir(A, X)
+	var/turf/T = X.loc
+	var/turf/temp = X.loc
+	for(var/x in 0 to max_distance)
+		temp = get_step(T, facing)
+		if(facing in diagonals) // check if it goes through corners
+			var/reverse_face = reverse_dir[facing]
+			var/turf/back_left = get_step(temp, turn(reverse_face, 45))
+			var/turf/back_right = get_step(temp, turn(reverse_face, -45))
+			if((!back_left || back_left.density) && (!back_right || back_right.density))
+				break
+		if(!temp || temp.density || temp.opacity)
+			break
+
+		var/blocked = FALSE
+		for(var/obj/structure/S in temp)
+			if(S.opacity || ((istype(S, /obj/structure/barricade) || istype(S, /obj/structure/machinery/door)) && S.density))
+				blocked = TRUE
+				break
+		if(blocked)
+			to_chat(X, SPAN_XENOWARNING("You can't reach [targetXeno] with your resin retrieval hook!"))
+			return
+
+		T = temp
+
+		if(T in turflist)
+			break
+
+		turflist += T
+		facing = get_dir(T, A)
+		telegraph_atom_list += new /obj/effect/xenomorph/xeno_telegraph/green(T, windup)
+
+	if(!length(turflist))
+		to_chat(X, SPAN_XENOWARNING("You don't have any room to do your retrieve!"))
+		return
+
+	X.visible_message(SPAN_XENODANGER("[X] prepares to fire its resin retrieval hook at [A]!"), SPAN_XENODANGER("You prepare to fire your resin retrieval hook at [A]!"))
+	X.emote("roar")
+
+	var/throw_target_turf = get_step(X.loc, facing)
+	var/turf/behind_turf = get_step(X.loc, reversefacing)
+	if(!(behind_turf.density))
+		throw_target_turf = behind_turf
+
+	X.frozen = TRUE
+	X.update_canmove()
+	if(windup)
+		if(!do_after(X, windup, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, null, null, FALSE, 1, FALSE, 1))
```

Comment:
```suggestion
		if(!do_after(X, windup, INTERRUPT_NO_NEEDHAND, BUSY_ICON_HOSTILE, null, null, FALSE, 1, FALSE, 1))
```
Can possibly do this better/more readable.

Reply:
improved